### DISCUSSION TITLE
Timeline for error on direct `setup.py` invokations?

### DISCUSSION THREAD

**Discussion by astrojuanlu** [pypa/setuptools#2904] [General]
I'm thrilled to read that setuptools has deprecated direct `setup.py` invokations, thanks a lot!

Is there a timeline for this to _raise an error_?

---

For the record, I already proposed that Read the Docs caps the setuptools version https://github.com/readthedocs/readthedocs.org/pull/8711 to guard ourselves of a future release of setuptools that breaks backwards compatibility.

**Comment by webknjaz** [comment]
cc @jaraco

**Comment by jaraco** [comment]
Thanks for asking.

Given that these invocations are so pervasive, I would expect the deprecation to be around for some time. It likely will not be before pip releases `--use-pep517` by default (and thus its implicit invocation of `setup.py egg_info/install`) and that's been in the wild for some time.

For that reason, it's possible, even likely, that Setuptools will release backward-incompatible releases for other reasons (at least one being https://github.com/pypa/setuptools/pull/2896, so use that to inform your decision about pinning Setuptools. Still, I would not be offended if you choose to pin Setuptools and explicitly unpin at each major version release (where appropriate).

If it would help give some predictability, I'd be happy to put a comment next to the deprecation warning to guard against removal before a certain date.

**Reply by astrojuanlu** [reply]
Thanks a lot for the prompt reply @jaraco, sounds good to me. xref https://github.com/pypa/pip/issues/9175 for the transition to `--use-pep517` by default.

For Read the Docs, we will probably pin at 58.3.0, since users can install newer versions if they want anyway. This will only affect the defaults, and avoid unexpected breakage even if we miss an important setuptools change.
